### PR TITLE
Docs: Example fix for `no-extra-parens` rule (fixes #3527)

### LINKED ISSUE #3527: `no-extra-parens` possible documentation issue
In the [rule docs](http://eslint.org/docs/rules/no-extra-parens), this is not supposed to give an error, and it doesn't:

```js
/*eslint no-extra-parens: 2*/

(function(){} ? a() : b())

(/^a$/).test(x);
```

However, if a semicolon is added to the end (or if the second statement is removed), it does give an error.
```js
/*eslint no-extra-parens: 2*/

(function(){} ? a() : b());

(/^a$/).test(x);
```
or
```js
/*eslint no-extra-parens: 2*/

(function(){} ? a() : b())
```

```
3:1  error  Gratuitous parentheses around expression  no-extra-parens
```

I'm thinking that example should have a semicolon added and moved to the "considered warnings" section, yes?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26185964-no-extra-parens-possible-documentation-issue?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: docs/rules/no-extra-parens.md [hunk 1]
```diff
@@ -34,8 +36,6 @@ The following patterns are not considered problems:
 
 ({}.toString.call());
 
-(function(){} ? a() : b())
```
Comment (michaelficarra) [inline docs/rules/no-extra-parens.md]: The better change would be to move the closing parenthesis after the function. This example was meant to illustrate the fact that some values may be parenthesised for reasons other than precedence.

Comment (gyandeeps) [inline docs/rules/no-extra-parens.md]: Are you recommending this:
```js
(function(){}) ? a() : b();
```

Comment (michaelficarra) [inline docs/rules/no-extra-parens.md]: Yes.

Comment (gyandeeps) [inline docs/rules/no-extra-parens.md]: Done. Thanks

**Pull request merged**